edx/edx-enterprise
# [ENT-49] Add catalog field that searches the API for catalog names

This pull request adds a catalog field to the EnterpriseCustomer, along with a form used in the admin view to dynamically grab the names of the available catalogs from a linked Course Discovery instance.

**JIRA tickets**: Implements ENT-49.

**Discussions**: See ENT-49

**Dependencies**: None

**Screenshots**:
![screen shot 2016-11-10 at 7 49 15 pm](https://cloud.githubusercontent.com/assets/7773758/20200423/1efe5754-a77f-11e6-9b03-8563d81b2db6.png)

**Sandbox URL**: TBD

**Testing instructions**:

1. Clone edx/course-discovery.
2. Configure EdX OpenID Connect [per the instructions on this page](https://open-edx-course-catalog.readthedocs.io/en/latest/getting_started.html).
5. Set `COURSE_CATALOG_API_URL` in your devstack LMS env to use your host's local IP address.
2. Run `make open-devstack`
4. Inside the course discovery devstack:

        ./manage.py create_or_update_partner --code='abc' --name='ABC Partner' --oidc-url-root=http://HOST_IP:8000/oauth2 --oidc-key='OIDC_KEY' --oidc-secret='OIDC_SECRET' --courses-api-url=http://HOST_IP:8000/api/courses/v1/

        ./manage.py refresh_course_metadata

        ./manage.py createsuperuser #Create the user course_discovery_worker

5. Once you've run the above commands, close out of the docker guest, and run `make start-devstack`.
6. Navigate to http://localhost:18381/admin, log in with `course_discovery_worker` and set up a couple course catalogs. You don't need to test the queries.
2. Install this fork of edx-enterprise into your devstack, adding it as an enabled application.
1. In your devstack, run `./manage.py lms migrate --settings=devstack`.
1. In your devstack run `./manage.py lms createsuperuser`. Add `user course_discovery_worker`.
3. Go to the EnterpriseCustomer admin panel, using the `course_discovery_worker` user, create a new EnterpriseCustomer, and observe that the list of available catalogs matches the catalogs listed in your course discovery stack.

**Author notes and concerns**:

1. Remove catalog option needs to be added
2. Tests need to be created 

**Reviewers**
- [x] @asadiqbal08 
- [ ] @saleem-latif 

**Merge checklist:**

**Post merge:**
